File: VB/GridView.CustomGrouping/Controllers/GridViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports GridView.CustomGrouping.Models

Namespace GridView.CustomGrouping.Controllers
	<HandleError> _
	Public Class GridViewController
		Inherits Controller
		Public Function Index() As ActionResult
			Return Grouping()
		End Function
		Public Function Grouping() As ActionResult
			Return View("Grouping", NorthwindDataProvider.GetProducts())
		End Function
		Public Function GroupingPartial() As ActionResult
			Return PartialView("GroupingPartial", NorthwindDataProvider.GetProducts())
		End Function
	End Class
End Namespace

File: CS/GridView.CustomGrouping/Controllers/GridViewController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using GridView.CustomGrouping.Models;

namespace GridView.CustomGrouping.Controllers {
    [HandleError]
    public class GridViewController : Controller {
        public ActionResult Index() {
            return Grouping();
        }
        public ActionResult Grouping() {
            return View("Grouping", NorthwindDataProvider.GetProducts());
        }
        public ActionResult GroupingPartial() {
            return PartialView("GroupingPartial", NorthwindDataProvider.GetProducts());
        }
    }
}
